addons/hr_timesheet/models/hr_timesheet.py

```diff
@@ -65,6 +72,65 @@ def write(self, values):
         self.filtered(lambda t: t.project_id)._timesheet_postprocess(values)
         return result
 
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form', toolbar=False, submenu=False):
+        """ Set the correct widget for `unit_amount` field on tree and form views, depending on company UoM configuration (`float_toggle` for Days, `float_time` for Hours, ...) """
+        result = super(AccountAnalyticLine, self).fields_view_get(view_id=view_id, view_type=view_type, toolbar=toolbar, submenu=submenu)
```

Comment:
If can find attribute company_time_unit, replace that attribute by the relevant wiget.


                    <field name="unit_amount" string="Time (Hours)" sum="Total time" company_time_unit="1"/>